### PR TITLE
Support drop table and view in new Parser and new Analyzer

### LINKED ISSUE #4042: New Parser and Analyzer support table and partition related statements【Easy】
# Requirement Description
The new Parser and the new Analyzer support table and partition related statements

# Mentor
@HangyuanLiu 

# Registration procedure

**Step 1.** Add a comment when you sign up. The comment can be a brief self-introduction and reason for signing up, including what you do, what you're good at, what you're interested in and your technology stack, and how familiar you are with StarRocks. Example:

> I pick! My name is Johnny. I am from StarRocks and I have been working on database kernel development for 1 year. I am interested in OLAP, good at C++ programming, and have been using StarRocks for 6 months. I chose to solve this issue because I am familiar with execution engine.

**Step 2.** On your WeChat, add the StarRocks community assistant "**_StarRocks-1_**" as a WeChat friend and leave your GitHub ID. The assistant will invite you to a WeChat group based on the issue you picked.

**Step 3.** The mentor will communicate with all the participants who picked this issue and assign the task to the best match. 

# Before you start
1. Read the StarRocks [GitHub workflow](https://github.com/StarRocks/community/blob/main/Contributors/guide/workflow.md) carefully, which provides the configurations of BE and FE and describes how to submit a pull request.
2. To learn more about StarRocks, read [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).

# Pull request requirements

1. After you complete code development, upload the code to the StarRocks code repo as a pull request (PR).

  - PR title format: **[SR GeekCamp] + PR title**
  - Labels ：[Geek Camp 2022/feature]
2. After you submit your PR, comment on the status of your PR submission as requested under the Pinned Issue [[StarRocks geek Camp 2022] ](https://github.com/StarRocks/starrocks/issues/4008)page.

  - Comment format: **"I have finished Issue #" + PR link id**

### PR DESCRIPTION
**Pull request by zhuxt2015**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4042

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 33 / 38 (86.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|18|22|81.82%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropTableStmt.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 1]
```diff
@@ -154,7 +155,8 @@ public static boolean supportedByNewParser(StatementBase statement) {
                 || statement instanceof QueryStmt
                 || statement instanceof QueryStatement
                 || statement instanceof ShowDbStmt
-                || statement instanceof ShowTableStmt;
+                || statement instanceof ShowTableStmt
+                || statement instanceof DropTableStmt;
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:159]: Add a new Statement in lexicographical order

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 2]
```diff
@@ -171,6 +173,7 @@ public static boolean supportedByNewAnalyzer(StatementBase statement) {
                 || statement instanceof ShowTableStmt
                 || statement instanceof ShowTableStatusStmt
                 || statement instanceof ShowVariablesStmt
-                || statement instanceof ShowWorkGroupStmt;
+                || statement instanceof ShowWorkGroupStmt
+                || statement instanceof DropTableStmt;
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:177]: Add a new Statement in lexicographical order

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 3]
```diff
@@ -35,6 +35,8 @@ statement
     | ALTER VIEW qualifiedName
         ('(' columnNameWithComment (',' columnNameWithComment)* ')')?
         AS queryStatement                                                               #alterView
+    | DROP TABLE (IF EXISTS)? qualifiedName (FORCE)?                                    #dropTable
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:38]:
```suggestion
    | DROP TABLE (IF EXISTS)? qualifiedName FORCE?                                    #dropTable
```

File: fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java [hunk 4]
```diff
@@ -49,6 +49,7 @@
 import com.starrocks.qe.VariableMgr;
 import com.starrocks.sql.StatementPlanner;
 import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.analyzer.ViewDefBuilder;
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java:52]: Remove useless imports

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 5]
```diff
@@ -97,6 +98,12 @@ public Void visitShowStatement(ShowStmt statement, ConnectContext session) {
             return null;
         }
 
+        @Override
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:101]: It is a good design to unify Drop related statement into one DropAnalyzer. But we don't need a separate visitDropStatement, it is enough to handle it in visitDropTableStmt

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        @Override
+        public Void visitDropTableStmt(DropTableStmt statement, ConnectContext context) {
+            String db = statement.getDbName();
+            db = getFullDatabaseName(db, context);
+            statement.setDb(db);
+            String tableName = statement.getTableName();
+            if (Strings.isNullOrEmpty(tableName)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_NO_TABLES_USED);
+            }
+            // check access
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java:35]: Corresponding Priv needs to be added in PrivilegeChecker

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    static class DropStmtAnalyzerVisitor extends AstVisitor<Void, ConnectContext> {
+        public void analyze(DdlStmt statement, ConnectContext session) {
+            visit(statement, session);
+        }
+
+        @Override
+        public Void visitDropTableStmt(DropTableStmt statement, ConnectContext context) {
+            String db = statement.getDbName();
+            db = getFullDatabaseName(db, context);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java:29]: use normalizationTableName

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 31 / 41 (75.61%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|17|20|85.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropTableStmt.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 31 / 40 (77.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|17|20|85.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropTableStmt.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 35 / 40 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|17|20|85.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropTableStmt.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            String dbName = statement.getDbName();
+            dbName = getFullDatabaseName(dbName, context);
+            statement.setDbName(dbName);
+            String tableName = statement.getTableName();
+            if (Strings.isNullOrEmpty(tableName)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_NO_TABLES_USED);
+            }
+            return null;
+        }
+        String getFullDatabaseName(String db, ConnectContext session) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java:35]: use normalizationTableName

Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/DropStmtAnalyzer.java:35]: use normalizationTableName

File: fe/fe-core/src/test/java/com/starrocks/analysis/DropTableStmtTest.java [hunk 9]
```diff
@@ -21,80 +21,58 @@
...
 import com.starrocks.qe.ConnectContext;
-import mockit.Expectations;
-import mockit.Mocked;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.utframe.UtFrameUtils;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
 public class DropTableStmtTest {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/analysis/DropTableStmtTest.java:32]: Can DropTableStmtTest.java be merged into AnalyzerDropTest.java?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 32 / 35 (91.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropTableStmt.java|10|10|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by Johnsonginati** [comment]
Nice work！

**Pull request merged**